delta-app/src/main/java/io/cdap/delta/store/RemoteStateStore.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.delta.store;
+
+import com.google.gson.Gson;
+import io.cdap.cdap.api.ServiceDiscoverer;
+import io.cdap.delta.api.DeltaPipelineId;
+import io.cdap.delta.app.DeltaWorkerId;
+import io.cdap.delta.app.OffsetAndSequence;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import javax.ws.rs.HttpMethod;
+
+/**
+ * Class to handle Rest Api calls to System Service [ Assessor ]
+ */
+
+public class RemoteStateStore implements StateStore {
+
+  private static final String BASE_PATH = "v1";
+  private static final String OFFSET_KEY = "offset";
+
+  private AssessmentServiceClient assessmentServiceClient;
+  private static final Gson GSON = new Gson();
+
+  public RemoteStateStore(ServiceDiscoverer context) {
+    this(new AssessmentServiceClient(context));
+  }
+
+  public RemoteStateStore(AssessmentServiceClient assessmentServiceClient) {
+    this.assessmentServiceClient = assessmentServiceClient;
+  }
+
+  @Override
+  public OffsetAndSequence readOffset(DeltaWorkerId id) {
+    String offsetAndSequenceJsonStr =
+      assessmentServiceClient.retryableApiCall(getMethodPath(id, OFFSET_KEY), HttpMethod.GET, null);
+    return GSON.fromJson(offsetAndSequenceJsonStr, OffsetAndSequence.class);
+  }
+
+  @Override
+  public void writeOffset(DeltaWorkerId id, OffsetAndSequence offset) {
+    String requestBody = GSON.toJson(offset);
+    assessmentServiceClient.retryableApiCall(getMethodPath(id, OFFSET_KEY), HttpMethod.PUT, requestBody);
+  }
+
+  @Override
+  public byte[] readState(DeltaWorkerId id, String key) {
+    String byteString = assessmentServiceClient.retryableApiCall(getMethodPathState(id, key), HttpMethod.GET, null);
+    return byteString.getBytes(StandardCharsets.UTF_8);
+  }
+
+  @Override
+  public void writeState(DeltaWorkerId id, String key, byte[] val) {
+    String valString = new String(val, StandardCharsets.UTF_8);
+    assessmentServiceClient.retryableApiCall(getMethodPathState(id, key), HttpMethod.PUT, valString);
+  }
+
+  @Override
+  public Long getLatestGeneration(String namespace, String pipelineName) {
+    return null;
```

Comment:
if this and getWorkerInstances() are never expected to be called, better to throw an UnsupportedOperationException instead of returning null